storage_service/locations/models/dataverse.py

```diff
@@ -17,55 +19,115 @@
+    space = models.OneToOneField("Space", to_field="uuid")
 
-class Dataverse(models.Model):
-    space = models.OneToOneField('Space', to_field='uuid')
-    host = models.CharField(max_length=256,
-        verbose_name=_l('Host'),
-        help_text=_l('Hostname of the Dataverse instance. Eg. apitest.dataverse.org'))
-    api_key = models.CharField(max_length=50,
-        verbose_name=_l('API key'),
-        help_text=_l('API key for Dataverse instance. Eg. b84d6b87-7b1e-4a30-a374-87191dbbbe2d'))
-    agent_name = models.CharField(max_length=50,
-        verbose_name=_l('Agent name'),
-        help_text=_l('Agent name for premis:agentName in Archivematica'))
-    agent_type = models.CharField(max_length=50,
-        verbose_name=_l('Agent type'),
-        help_text=_l('Agent type for premis:agentType in Archivematica'))
-    agent_identifier = models.CharField(max_length=256,
-        verbose_name=_l('Agent identifier'),
-        help_text=_l('URI agent identifier for premis:agentIdentifierValue in Archivematica'))
+    host = models.CharField(
+        max_length=256,
+        verbose_name=_l("Host"),
+        help_text=_l(
+            "Hostname of the Dataverse instance. Eg. apitest.dataverse.org"
+        ),
+    )
+    api_key = models.CharField(
+        max_length=50,
+        verbose_name=_l("API key"),
+        help_text=_l(
+            "API key for Dataverse instance. Eg. "
+            "b84d6b87-7b1e-4a30-a374-87191dbbbe2d"
+        ),
+    )
+    agent_name = models.CharField(
+        max_length=50,
+        verbose_name=_l("Agent name"),
+        help_text=_l("Agent name for premis:agentName in Archivematica"),
+    )
+    agent_type = models.CharField(
+        max_length=50,
+        verbose_name=_l("Agent type"),
+        help_text=_l("Agent type for premis:agentType in Archivematica"),
+    )
+    agent_identifier = models.CharField(
+        max_length=256,
+        verbose_name=_l("Agent identifier"),
+        help_text=_l(
+            "URI agent identifier for premis:agentIdentifierValue "
+            "in Archivematica"
+        ),
+    )
     # FIXME disallow string in space.path
 
     class Meta:
         verbose_name = _l("Dataverse")
-        app_label = 'locations'
+        app_label = "locations"
+
+    ALLOWED_LOCATION_PURPOSE = [Location.TRANSFER_SOURCE]
+
+    def get_query_value(self, key, path, dataset=False, default=None):
+        """Retrieve a value from a string and then remove it where it exists so
+        that we can recurse through it for all the values we need.
+        """
+        path = path.lower()
+        value = path.split(key.lower(), 1)
+        try:
+            value = value[1].strip()
+        except IndexError:
+            value = default
+        if value == "":
+            value = default
+        if value is None:
+            path = path.replace(key, "")
+        else:
+            if dataset:
+                return value, path.replace(key + value, "").strip()
```

Comment:
You could avoid the use of the dataset kwarg and treat all keys the same way if you did something like the following:

    +import re
     import zipfile
    
     # Core Django, alphabetical
    @@ -66,7 +67,7 @@ class Dataverse(URLMixin, models.Model):
    
         ALLOWED_LOCATION_PURPOSE = [Location.TRANSFER_SOURCE]
    
    -    def get_query_value(self, key, path, dataset=False, default=None):
    +    def get_query_value(self, key, path, default=None):
             """Retrieve a value from a string and then remove it where it exists so
             that we can recurse through it for all the values we need.
             """
    @@ -81,9 +82,8 @@ class Dataverse(URLMixin, models.Model):
             if value is None:
                 path = path.replace(key, "")
             else:
    -            if dataset:
    -                return value, path.replace(key + value, "").strip()
    -            path = path.replace(value, "").replace(key, "")
    +            path = re.sub(
    +                re.escape(key) + r"\s*" + re.escape(value), "", path)
             return value, path.strip()
    
         def get_query_and_subtree(self, path):
    @@ -95,7 +95,7 @@ class Dataverse(URLMixin, models.Model):
             browser. We can then retrieve the subtree and the query string which
             can be used to return a listing of datasets available in the Dataverse.
             """
    -        dataset, path = self.get_query_value("/", path, dataset=True)
    +        dataset, path = self.get_query_value("/", path)
             subtree, path = self.get_query_value("subtree:", path)
             query_string, path = self.get_query_value("query:", path, default="*")
             return dataset, subtree, query_string